src/explorer.api/Exploration/ExplorationRegistry.cs

```diff
@@ -52,17 +67,22 @@ public void CancelExploration(Guid id)
 
         private class Registration : IDisposable
         {
-            private readonly CancellationTokenSource tokenSource;
+            private readonly CancellationTokenSource tokenSource = new CancellationTokenSource();
 
             private bool disposedValue;
 
-            public Registration(Exploration exploration, CancellationTokenSource tokenSource)
+            public Registration(ExplorationLauncher launcher, ExploreParams requestData)
             {
-                Exploration = exploration;
-                this.tokenSource = tokenSource;
+                RequestData = requestData;
+                ValidationTask = Task.Run(async () =>
+                    Exploration = await launcher.ValidateAndLaunch(requestData, tokenSource.Token));
```

Comment:
I agree. 

I created the `Registration` initially to encapsulate all the metadata about the exploration (cancellationTokenSource, original request params) but then it kind of grew a bit too big 😄 .

I think the main issue here is that the parameter validation step needs to be added to the `Exploration` as you suggested (maybe we should even pull it into an `ExplorerComponent` that publishes the validated context and the `ExplorerInfo` or something similar. 

It will require some further refactoring though, since some of the functionality has dependencies on the `aircloak` project and the `explorer` project does not have access to `aircloak`.

Reply:
(I've created an issue #261 )